OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -404,9 +405,44 @@ public string DirectoryName(uint index)
 			return test;
 		}
 
+		public string GetFirstSubDirectory(string package)
```

Comment:
I'm deliberately ignoring the motivations because replacing the opaque file IDs with filenames is a good idea.